commands/mr/merge/mr_merge.go

```diff
@@ -187,8 +187,18 @@ func NewCmdMerge(f *cmdutils.Factory) *cobra.Command {
 			mrIID := mr.IID
 
 			err = retry.Do(func() error {
-				mr, err = api.MergeMR(apiClient, repo.FullName(), mrIID, mergeOpts)
-				if err != nil && (!opts.RebaseBeforeMerge || err.Error() != "Branch cannot be merged") {
+				var resp *gitlab.Response
+				mr, resp, err = api.MergeMR(apiClient, repo.FullName(), mrIID, mergeOpts)
+				if err != nil {
+					// https://docs.gitlab.com/ee/api/merge_requests.html#accept-mr
+					// `406` is the documented status code we will receive if the
+					// branch cannot be merged, also make sure we are rebasing
+					// before we merge as this is the specific situation we are
+					// avoiding
+					if opts.RebaseBeforeMerge && resp.StatusCode == 406 {
+						return err
+					}
```

Comment:
It will be best to check for only the status code. The user might have already initiated a rebase either on the web UI or used `glab mr rebase` before running `glab mr merge` without using the `--rebase` flag so a rebase could be in progress

```suggestion
					if resp.StatusCode == 406 {
						return err
					}
```

Reply:
Thanks, I didn't consider that the user can do a rebase via an external source before.